app/scripts/lib/snap-keyring/utils/getSnapName.test.ts

```diff
@@ -0,0 +1,90 @@
+import { SubjectMetadataController } from '@metamask/permission-controller';
+import { ControllerMessenger } from '@metamask/base-controller';
+import { getSnapName } from './getSnapName';
+
+const MOCK_SNAP_NAME = 'snap.test';
+const MOCK_SNAP_PREFIX = 'local';
+const MOCK_SNAP_ORIGIN = `${MOCK_SNAP_PREFIX}:${MOCK_SNAP_NAME}`;
+
+function setupControllerMessenger() {
+  const controllerMessenger = new ControllerMessenger();
+
+  // The SubjectMetadataController also requires some methods from PermissionController
+  controllerMessenger.registerActionHandler(
+    'PermissionController:hasPermissions',
+    // Not important for our tests
+    (_) => true as never,
+  );
+
+  const subjectMetadata = [
+    {
+      origin: MOCK_SNAP_ORIGIN,
+      name: MOCK_SNAP_NAME,
+    },
+  ];
+  const subjectMetadataController = new SubjectMetadataController({
+    messenger: controllerMessenger.getRestricted({
+      name: 'SubjectMetadataController',
+      allowedActions: [
+        'PermissionController:hasPermissions' as unknown as never,
+      ],
+    }),
+    subjectCacheLimit: subjectMetadata.length,
+  });
+  subjectMetadata.forEach((x) =>
+    subjectMetadataController.addSubjectMetadata(x),
+  );
+
+  return controllerMessenger;
+}
+
+function getSnapKeyringBuilderMessenger(
+  controllerMessenger: any,
+  allowedActions: string[],
+) {
+  return controllerMessenger.getRestricted({
+    name: 'SnapKeyringBuilder',
+    allowedActions,
+  });
+}
+
+describe('getSnapName', () => {
+  const controllerMessenger = setupControllerMessenger();
+  const messenger = getSnapKeyringBuilderMessenger(controllerMessenger, [
+    'SubjectMetadataController:getSubjectMetadata',
+  ]);
+
+  it('should returns the snap name', () => {
+    expect(getSnapName(messenger, MOCK_SNAP_ORIGIN)).toBe(MOCK_SNAP_NAME);
+  });
+
+  it('should returns the snap name even if subject metadata is not known', () => {
+    const snapName = 'unknown-snap.test';
+    const snapId = `${MOCK_SNAP_PREFIX}:${snapName}`;
+
+    expect(getSnapName(messenger, snapId)).toBe(snapName);
+  });
+
+  it('null should be null', () => {
+    // In our case we should probably never get a `null` origin (== snapId), but for the sake
+    // of completeness we make sure this is a well defined behavior
+    expect(getSnapName(messenger, null as unknown as string)).toBe(null);
+  });
+
+  it('undefined should be null', () => {
+    // Same here
+    expect(getSnapName(messenger, undefined as unknown as string)).toBe(null);
+  });
+
+  it('cannot get subject metadata will raise', () => {
```

Comment:
The name of the tests are a bit weird, for example, you should read like, `it undefined should be null`, you can rephrase like `it should return null if snapId is undefined` -> `it('should return null if snapId is undefined')`

Reply:
Oh yeah you right, I'm pretty sure I wanted to change those and completely forgot 🙈 Let me fix that!